storage/range_command.go

```diff
@@ -234,7 +263,7 @@ func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args *
 		}
 	} else {
 		// The transaction doesn't exist yet on disk; use the supplied version.
-		reply.Txn = gogoproto.Clone(args.Txn).(*proto.Transaction)
+		reply.Txn = args.Txn
```

Comment:
Why did you remove the clone? It looks deliberate (although if it was important I'd hope we have a test for it).